File: HyprMX-Android-ExampleApp/app-java/src/main/java/com/hyprmx/android/exampleapp/Utilities.java
package com.hyprmx.android.exampleapp;

import android.content.Context;
import android.util.TypedValue;
import android.view.View;

import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

public class Utilities {

  public static int dpToPx(Context context, int dp) {
    return (int) TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, dp, context.getResources().getDisplayMetrics());
  }

  public static void applyWindowInsets(View view) {
    applyWindowInsets(view, WindowInsetsCompat.Type.systemBars() | WindowInsetsCompat.Type.displayCutout());
  }

  public static void applyWindowInsets(View view, int insetTypes) {
    ViewCompat.requestApplyInsets(view);

    ViewCompat.setOnApplyWindowInsetsListener(view, (v, insets) -> {
      Insets bars = insets.getInsets(insetTypes);

      v.setPadding(bars.left, bars.top, bars.right, bars.bottom);
      return WindowInsetsCompat.CONSUMED;
    });
  }
}

File: HyprMX-Android-ExampleApp/app-kotlin/src/main/java/com/hyprmx/android/exampleapp/Utilities.kt
package com.hyprmx.android.exampleapp

import android.content.Context
import android.util.TypedValue
import android.view.LayoutInflater
import android.view.View
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import androidx.core.view.updatePadding

fun Context.dpToPx(dp: Int): Int =
  TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, dp.toFloat(), resources.displayMetrics)
    .toInt()

fun Context.layoutInflater(): LayoutInflater = LayoutInflater.from(this)

fun View.applyWindowInsets(insetsType: Int = WindowInsetsCompat.Type.systemBars() or WindowInsetsCompat.Type.displayCutout()) {
  ViewCompat.requestApplyInsets(this)

  ViewCompat.setOnApplyWindowInsetsListener(this) { view, insets ->
    val bars = insets.getInsets(insetsType)

    view.updatePadding(
      left = bars.left,
      top = bars.top,
      right = bars.right,
      bottom = bars.bottom,
    )

    WindowInsetsCompat.CONSUMED
  }
}
